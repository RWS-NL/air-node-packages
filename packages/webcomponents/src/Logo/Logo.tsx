import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import React, { FC, memo, SVGProps } from 'react';
import css from './Logo.scss';
import { ReactComponent as LogoSVG } from './logo.svg';

export interface LogoProps extends SVGProps<any> {
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: string;
  /** Custom CSS classes to pass to the button */
  customclasses?: string | string[];
}

/**
 * Creates a logo using pre-defined Rijkswaterstaat SVG
 * @example
 * <Logo height={200} width={500}/>
 */
export const Logo: FC<LogoProps> = props => <LogoSVG {...props} className={classnames(css.logo, css.customclasses)} />;

export default memo(Logo);