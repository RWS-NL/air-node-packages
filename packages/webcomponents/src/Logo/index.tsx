import { SxProps } from '@mui/material';
import React, { memo, SVGProps } from 'react';
import { ReactComponent as LogoSVG } from './logo.svg';

export interface LogoProps extends SVGProps<any> {
  /** Any additional CSSProperties to pass to the component */
  sx?: SxProps;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: string;
}

/**
 * Creates a logo using pre-defined Rijkswaterstaat SVG
 * @example
 * ```jsx
 *
 * <Logo height={125} width={314}/>
 *
 * ```
 */
export const Logo = memo((props: LogoProps) => <LogoSVG {...props} />);
