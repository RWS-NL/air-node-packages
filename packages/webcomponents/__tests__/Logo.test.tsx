import { Logo } from '@src/Logo';
import { shallow } from 'enzyme';
import React from 'react';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const logo = shallow(<Logo />);
    expect(logo).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const logo = shallow(<Logo height={500} width={500} cursor='pointer' />);
    expect(logo).toMatchSnapshot();
  });
});
