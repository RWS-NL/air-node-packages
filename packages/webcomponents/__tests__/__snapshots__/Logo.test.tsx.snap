// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<IconMock
  className="logo customclasses"
  cursor="pointer"
  height={500}
  width={500}
/>
`;

exports[`Snapshot Testing Required Props 1`] = `
<IconMock
  className="logo customclasses"
/>
`;
